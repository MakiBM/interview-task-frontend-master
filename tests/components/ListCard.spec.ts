import { describe, it, expect, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import Component from '../../src/components/ListCard.vue'

describe('ListCard Snapshot Tests', () => {
  it('matches the snapshot without a selected item', () => {
    const wrapper = mount(Component, {
      props: {
        items: ['Item 1', 'Item 2', 'Item 3'],
        selected: null, // No item selected
      },
      slots: {
        default: '<h5>Card Header</h5>', // Provide slot content
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('matches the snapshot with a selected item', () => {
    const wrapper = mount(Component, {
      props: {
        items: ['Item 1', 'Item 2', 'Item 3'],
        selected: 'Item 2', // Pre-selected item
      },
      slots: {
        default: '<h5>Card Header</h5>', // Provide slot content
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
