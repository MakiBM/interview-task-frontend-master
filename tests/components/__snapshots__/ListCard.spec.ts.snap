// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ListCard Snapshot Tests > matches the snapshot with a selected item 1`] = `
"<div class="card border-0 h-100" data-v-683929d4="">
  <div class="card-header" data-v-683929d4="">
    <div class="p-3 px-2" data-v-683929d4="">
      <!-- Card header -->
      <h5>Card Header</h5>
    </div>
  </div>
  <div class="card-body" data-v-683929d4="">
    <ul class="list-group list-group-flush" data-v-683929d4="">
      <li class="list-group-item list-group-item-action py-3" role="button" data-v-683929d4="">
        <p class="px-2 mb-0" data-v-683929d4=""><small data-v-683929d4="">Item 1</small></p>
      </li>
      <li class="list-group-item list-group-item-action py-3 text-primary fw-medium" role="button" data-v-683929d4="">
        <p class="px-2 mb-0" data-v-683929d4=""><small data-v-683929d4="">Item 2</small></p>
      </li>
      <li class="list-group-item list-group-item-action py-3" role="button" data-v-683929d4="">
        <p class="px-2 mb-0" data-v-683929d4=""><small data-v-683929d4="">Item 3</small></p>
      </li>
    </ul>
  </div>
</div>"
`;

exports[`ListCard Snapshot Tests > matches the snapshot without a selected item 1`] = `
"<div class="card border-0 h-100" data-v-683929d4="">
  <div class="card-header" data-v-683929d4="">
    <div class="p-3 px-2" data-v-683929d4="">
      <!-- Card header -->
      <h5>Card Header</h5>
    </div>
  </div>
  <div class="card-body" data-v-683929d4="">
    <ul class="list-group list-group-flush" data-v-683929d4="">
      <li class="list-group-item list-group-item-action py-3" role="button" data-v-683929d4="">
        <p class="px-2 mb-0" data-v-683929d4=""><small data-v-683929d4="">Item 1</small></p>
      </li>
      <li class="list-group-item list-group-item-action py-3" role="button" data-v-683929d4="">
        <p class="px-2 mb-0" data-v-683929d4=""><small data-v-683929d4="">Item 2</small></p>
      </li>
      <li class="list-group-item list-group-item-action py-3" role="button" data-v-683929d4="">
        <p class="px-2 mb-0" data-v-683929d4=""><small data-v-683929d4="">Item 3</small></p>
      </li>
    </ul>
  </div>
</div>"
`;
